src/data/sign.lean

```diff
@@ -202,3 +201,58 @@ def sign_hom : α →*₀ sign_type :=
                mul_neg_of_pos_of_neg, mul_pos] }
 
 end linear_ordered_ring
+
+namespace int
+
+@[simp] lemma sign_eq_sign (n : ℤ) : n.sign = _root_.sign n :=
+begin
+  obtain ((_ | _) | _) := n,
+  { exact congr_arg coe sign_zero.symm },
+  { exact congr_arg coe (sign_pos $ int.succ_coe_nat_pos _).symm },
+  { exact congr_arg coe (_root_.sign_neg $ neg_succ_lt_zero _).symm }
+end
+end int
+
+open finset nat
+open_locale big_operators
+
+variables [decidable_eq α] [nonempty α]
+
+-- TODO: Inlining this yields an app-builder exception
+lemma exists_signed_sum_aux {n : ℕ} (sgn : ℕ → sign_type) (b : α) [decidable_eq α] {f : α → ℤ}
+  ⦃a : α⦄ (g : ℕ → α) (i : ℕ) :
+   ite ((range (n - (f a).nat_abs)).piecewise g (λ _, a) i = b)
+        ((range (n - (f a).nat_abs)).piecewise sgn (λ _, sign (f a)) i : ℤ) 0 =
+    (range (n - (f a).nat_abs)).piecewise (λ j, ite (g j = b) ↑(sgn j) 0)
+        (λ j, ite (a = b) ↑(sign (f a)) 0) i :=
+by { unfold piecewise, split_ifs; refl }
+
+lemma exists_signed_sum (s : finset α) (n : ℕ) (f : α → ℤ) (hn : ∑ i in s, (f i).nat_abs ≤ n) :
+  ∃ (sgn : ℕ → sign_type) (g : ℕ → α), (∀ i, g i ∉ s → sgn i = 0) ∧
+    ∀ a ∈ s, (∑ i in range n, if g i = a then (sgn i : ℤ) else 0) = f a :=
```

Comment:
This statement is quite hard for me to parse. Is there not a much simpler way to state this?

Reply:
Not sure...
Here's an example:
```
1 + 2 - 4
```
becomes
```
1 + 0 + 0 + 0 + 0 + 0 + 0
0 + 1 + 1 + 0 + 0 + 0 + 0
0 + 0 + 0 - 1 - 1 - 1 - 1
```

Reply:
Can you elaborate on what `sgn` and `g` are for that example? Can we define them constructively?

Reply:
In that case,
```
sgn: 1 1 1 -1 -1 -1 -1
g:   0 1 1  2  2  2  2
```
Yes, we can define them constructively, and I did try, but it's a real pain to keep track of `sgn` and `g` simultaneously (involves a yucky `list.join` that I didn't manage to get around).

Reply:
Can you link to where lte uses this?

Reply:
[Here](https://github.com/leanprover-community/lean-liquid/blob/master/src/free_pfpng/epi.lean#L503-L542)